[Device-capability] [Create and update document][1452622] (#2656)
MicrosoftDocs/msteams-docs: msteams-platform/concepts/device-capabilities/native-device-permissions.md

---
title: Request device permissions for your Microsoft Teams app
description: How to update your app manifest in order to request access to native features that usually require user consent
keywords: teams apps development
---

# Request device permissions for your Microsoft Teams app

Enrich your app with features that require access to native device functionality, such as:

> [!div class="checklist"]
>
> * Camera
> * Microphone
> * Gallery
> * Location

[!NOTE] 

> * To integrate your native device media capabilities within Teams mobile app, see [Media capabilities in Teams.](mobile-camera-image-permissions.md)

Reviewer: Not sure if this line is needed here. Also, please fix the Note tag.

* The device permissions work in the same way for all app constructs, such as tabs or messaging extensions.

> [!IMPORTANT]
>

> > * Support for `camera`, `gallery`, and `microphone` is enabled through [**selectMedia API**](/javascript/api/@microsoft/teams-js/media?view=msteams-client-js-latest#selectMedia_MediaInputs___error__SdkError__attachments__Media_______void_&preserve-view=true). For single image capture you may use [**captureImage API**](/javascript/api/@microsoft/teams-js/microsoftteams?view=msteams-client-js-latest#captureimage--error--sdkerror--files--file-------void-&preserve-view=true).

Reviewer: @Rajeshwari-v, is there a reason why this important is here? Can we move it to a some other section? It is breaking the flow of the page. Lets discuss.

You must leverage the  to get appropriate HTML5 or Teams API, to display a prompt for getting a consent to access device permissions. 

> [!IMPORTANT]
> * Support for `camera`, `gallery`, and `microphone` is enabled through [**selectMedia API**](/javascript/api/@microsoft/teams-js/media?view=msteams-client-js-latest#selectMedia_MediaInputs___error__SdkError__attachments__Media_______void_&preserve-view=true). Use [**captureImage API**](/javascript/api/@microsoft/teams-js/microsoftteams?view=msteams-client-js-latest#captureimage--error--sdkerror--files--file-------void-&preserve-view=true) for a single image capture.
> * Support for `location` is enabled through [**getLocation API**](/javascript/api/@microsoft/teams-js/location?view=msteams-client-js-latest#getLocation_LocationProps___error__SdkError__location__Location_____void_&preserve-view=true). You must use this `getLocation API` for location, as [**geolocation API**](../../resources/schema/manifest-schema.md#devicepermissions) is currently not fully supported on all desktop clients.

For example:
 * To prompt the user to access their location you must call `getCurrentPosition()`:

    ```Javascript

>     navigator.geolocation.getCurrentPosition    (function (position) { /*... */ });

Reviewer 2: ```suggestion
    navigator.geolocation.getCurrentPosition(function (position) { /*... */ });
```

    ```

>  * To use the camera on desktop or web, Teams displays a permission prompt when you call `getUserMedia()`:

Reviewer 2: ```suggestion
 * To prompt user to access their camera on desktop or web you must call `getUserMedia()`:
```

});
```

## Use Teams APIs to get device permissions

Leverage appropriate HTML5 or Teams API, to display a prompt for getting consent to access device permissions.

> [!IMPORTANT]
> * Support for `camera`, `gallery`, and `microphone` is enabled through [**selectMedia API**](/javascript/api/@microsoft/teams-js/media?view=msteams-client-js-latest#selectMedia_MediaInputs___error__SdkError__attachments__Media_______void_&preserve-view=true). Use [**captureImage API**](/javascript/api/@microsoft/teams-js/microsoftteams?view=msteams-client-js-latest#captureimage--error--sdkerror--files--file-------void-&preserve-view=true) for a single image capture.

> > * Support for `location` is enabled through [**getLocation API**](/javascript/api/@microsoft/teams-js/location?view=msteams-client-js-latest#getLocation_LocationProps___error__SdkError__location__Location_____void_&preserve-view=true). You must use this `getLocation API` for location, as [**geolocation API**](../../resources/schema/manifest-schema.md#devicepermissions) is currently not fully supported on all desktop clients.

Reviewer 2: Hyperlinks are giving 404 error. Please check again.

Author: The links will work once the PR is merged as these APIs are present in a different docset.

> > [!IMPORTANT]
> > * Support for `camera`, `gallery`, and `microphone` is enabled through [**selectMedia API**](/javascript/api/@microsoft/teams-js/media?view=msteams-client-js-latest#selectMedia_MediaInputs___error__SdkError__attachments__Media_______void_&preserve-view=true). Use [**captureImage API**](/javascript/api/@microsoft/teams-js/microsoftteams?view=msteams-client-js-latest#captureimage--error--sdkerror--files--file-------void-&preserve-view=true) for a single image capture.
> > * Support for `location` is enabled through [**getLocation API**](/javascript/api/@microsoft/teams-js/location?view=msteams-client-js-latest#getLocation_LocationProps___error__SdkError__location__Location_____void_&preserve-view=true). You must use this `getLocation API` for location, as [**geolocation API**](../../resources/schema/manifest-schema.md#devicepermissions) is currently not fully supported on all desktop clients.

Reviewer 2: ```suggestion
> [!IMPORTANT]
> * Support for `camera`, `gallery`, and `microphone` is enabled through [**selectMedia API**](/javascript/api/@microsoft/teams-js/media?view=msteams-client-js-latest#selectMedia_MediaInputs___error__SdkError__attachments__Media_______void_&preserve-view=true). Use [**captureImage API**](/javascript/api/@microsoft/teams-js/microsoftteams?view=msteams-client-js-latest#captureimage--error--sdkerror--files--file-------void-&preserve-view=true) for a single image capture.
> * Support for `location` is enabled through [**getLocation API**](/javascript/api/@microsoft/teams-js/location?view=msteams-client-js-latest#getLocation_LocationProps___error__SdkError__location__Location_____void_&preserve-view=true). You must use this `getLocation API` for location, as HTML5 geolocation API is currently not fully supported on Teams desktop client.
```
